src/hotspot/os/posix/os_posix.cpp

```diff
@@ -610,9 +610,11 @@ void os::print_jni_name_suffix_on(outputStream* st, int args_size) {
 
 bool os::get_host_name(char* buf, size_t buflen) {
   struct utsname name;
-  uname(&name);
-  jio_snprintf(buf, buflen, "%s", name.nodename);
-  return true;
+  if (uname(&name) == 0) {
+    jio_snprintf(buf, buflen, "%s", name.nodename);
+    return true;
+  }
```

Comment:
Else I suggest printing a warning and reporting the error using os::strerror

Reply:
Bikeshedding, I would just report false, but print a warning using UL, flag "os", with warning level. That will be shown by default, but this is strange enough that users should see this.